File: iOS开发范例实战宝典(基础篇)/按钮/06-旋转的图片/06-旋转的图片/ViewController.m
//
//  ViewController.m
//  06-旋转的图片
//
//  Created by coco on 16/5/30.
//  Copyright © 2016年 XHJ. All rights reserved.
//

#import "ViewController.h"
#import "HJRotationImage.h"
@interface ViewController ()
@property (weak, nonatomic) IBOutlet HJRotationImage *rotationView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.rotationView.imageV.image = [UIImage imageNamed:@"dog.jpg"];
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(10 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        [self.rotationView stopRotation];
    });
    
    HJRotationImage *imageV = [[HJRotationImage alloc] init];
    [self.view addSubview:imageV];
    imageV.frame = CGRectMake(50, 50, 100, 100);
    imageV.imageV.image = [UIImage imageNamed:@"dog.jpg"];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift练习/《iOS开发实战：图形图像、动画与多媒体卷（Swift语言版）》/第04章 动画技术/4.3__DynamicBehaviorSample/4.3__DynamicBehaviorSample/ViewController.swift
//
//  ViewController.swift
//  4.3__DynamicBehaviorSample
//
//  Created by coco on 16/7/5.
//  Copyright © 2016年 XHJ. All rights reserved.
//

import UIKit

//枚举和String关联了, 使用成员.rawValue 获取值
enum DynamicType:String {
    case GravityBehavior = "重力行为" //重力行为
    case CollisionBehavior = "碰撞行为" //碰撞行为
    case AttachmentBehavior = "吸附行为" //吸附行为
    case PushBehavior = "推行为"  //推行为
    case SnapBehavior = "甩行为" //甩行为
}


class ViewController: UITableViewController {

    var behaviorType:DynamicType = .GravityBehavior
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tableView.tableFooterView = UIView()
        
        //测试运动效果
        //x轴
        let motionEffectX = UIInterpolatingMotionEffect(keyPath: "center.x", type: UIInterpolatingMotionEffectType.TiltAlongHorizontalAxis)
        motionEffectX.maximumRelativeValue = 50
        motionEffectX.minimumRelativeValue = -50
        self.view.addMotionEffect(motionEffectX)
        
        let motionEffectY = UIInterpolatingMotionEffect(keyPath: "center.y", type: UIInterpolatingMotionEffectType.TiltAlongHorizontalAxis)
        motionEffectY.maximumRelativeValue = 50
        motionEffectY.minimumRelativeValue = -50
        self.view.addMotionEffect(motionEffectY)
        
        for (_, _) in "123456".characters.enumerate() {
            print("1")
        }
        let _ = "dfdf"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        
        switch indexPath.row {
        case 0: self.behaviorType = .GravityBehavior
        case 1: self.behaviorType = .CollisionBehavior
        case 2: self.behaviorType = .AttachmentBehavior
        case 3: self.behaviorType = .PushBehavior
        default: self.behaviorType = .SnapBehavior
        }
        
        self.performSegueWithIdentifier("Dynamic", sender: self.behaviorType.rawValue)
    }

    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        let destinationViewController:BehaviorController = segue.destinationViewController as! BehaviorController
        destinationViewController.behaviorType = self.behaviorType
        destinationViewController.title = self.behaviorType.rawValue

    }

    
}

